pdns/dnsdist-lua-actions.cc

```diff
@@ -939,6 +939,41 @@ class SetMacAddrAction : public DNSAction
   uint16_t d_code{3};
 };
 
+
+class SetEDNSOptionAction : public DNSAction
+{
+public:
+  // this action does not stop the processing
+  SetEDNSOptionAction(uint16_t code, const std::string& data) : d_code(code), d_data(data)
+  {}
+
+  DNSAction::Action operator()(DNSQuestion* dq, std::string* ruleresult) const override
+  {
+    if (dq->getHeader()->arcount) {
+      return Action::None;
```

Comment:
I wonder if we could easily refactor `slowRewriteQueryWithRecords` (`dnsdist-ecs.cc`) to handle that case as well?